reader_go18.go

```diff
@@ -149,6 +149,9 @@ func (r *GenericReader[T]) readRows(rows []T) (int, error) {
 		r.base.rowbuf = r.base.rowbuf[:nRequest]
 	}
 
+	schema := r.base.Schema()
+	columns := make([][]Value, len(schema.columns))
```

Comment:
Arguably these columns should be stored in the GenericReader object and reused across Read calls. That would help if Read is called many times with a relatively small rows slice. However, it's harder to know for sure that won't cause problems.